internal/controllers/addon/utils.go

```diff
@@ -319,13 +319,14 @@ func GetMonitoringFederationServiceMonitorEndpoints(addon *addonsv1alpha1.Addon)
 	}
 
 	return []monitoringv1.Endpoint{{
-		HonorLabels: true,
-		Port:        "9090",
-		Path:        "/federate",
-		Scheme:      "https",
-		Interval:    "30s",
-		TLSConfig:   tlsConfig,
-		Params:      map[string][]string{"match[]": matchParams},
+		BearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token",
+		HonorLabels:     true,
+		Port:            addon.Spec.Monitoring.Federation.PortName,
+		Path:            "/federate",
```

Comment:
I dont think Path is needed

Reply:
Looking at the `ServiceMonitorSpec` CRD, there is no default for Path. I think it is safe to put `/federate` and acts as documentation: https://github.com/prometheus-operator/prometheus-operator/blob/8e6de29b4c1be9cc2f322528c0d1f1b2492c4a32/pkg/apis/monitoring/v1/types.go#L1106

Reply:
FYI, I read this https://coreos.slack.com/archives/C0VMT03S5/p1639421772076600?thread_ts=1639407574.071400&cid=C0VMT03S5 from monitoring team